src/com/google/javascript/jscomp/Es6ConvertSuper.java

```diff
@@ -109,95 +114,100 @@ private boolean isInterface(Node classNode) {
   }
 
   private void visitSuper(Node node, Node parent) {
-    Node enclosingCall = parent;
-    Node potentialCallee = node;
-    if (!parent.isCall()) {
-      enclosingCall = parent.getParent();
-      potentialCallee = parent;
+    Node exprRoot = node;
+
+    if (exprRoot.getParent().isGetElem() || exprRoot.getParent().isGetProp()) {
```

Comment:
An internal test system found that changing this to be `if (true) {` didn't cause any test failures.
This implies a missing test and made me realize it's not as clear as it should be what's happening here.

Could this logic be clarified to something like this?

```java
private void visitSuper(Node superNode, Node parent) {
  checkState(superNode.isSuper());
  if (parent.isCall()) {
    // super(...)
    visitBareSuperCall(superNode, parent);
  } else if (parent.isGetProp()) {
    checkState(parent.getFirstChild() == superNode);
    // super.something
    if (parent.getParent().isCall()) {
      // super.something(...)
      visitSuperDotSomethingCall(superNode, parent);
    } else {
      // super.something used in some other way
      // report not supported yet error
    }
  } else if (parent.isGetElem()) {
    checkState(superNode == parent.getFirstChild());
    // super[...]
    // report not supported yet error
  } else if (parent.isNew()) {
    // new super(...)
    // invalid use of super
  } else {
    // some other use of super we don't support yet
  }
}
```